docs/components/docgen.js

```diff
@@ -46,8 +41,37 @@ export default async function docgen({
       // Remove images from the description
       .replace(/!\[(.*?)\][[(].*?[\])]/g, '');
   }
-
   return parsed;
+};
+
+export default async function docgen({
+  componentName,
+  alternativeSubdirectory = '',
+}: {|
+  componentName: string,
+  alternativeSubdirectory?: string,
+|}): Promise<DocGen> {
+  const filePath = getFilePath(componentName, alternativeSubdirectory);
+  const contents = await fs.promises.readFile(filePath, 'utf-8');
+  return getParsedContent(contents);
+}
+
+export async function multipledocgen({
```

Comment:
It looks like these two functions could be combined into one? The only difference seems to be if `componentName` is a string or array of strings, and if the return value is a DocGen promise or an array of DocGen promises. The logic of `multipleDocGen` looks like it could work for single or multiple components?

Reply:
I tried to have a single function but had to change API for function and had issues with Flowtyping it. So kept it simple. Also, we don't know how long we are going to use these, as it seems they might end up going into different pages,

Reply:
That's a good point!

Reply:
Refactored it to a map. I played  around with reduce but Promise.all  neews iterable  so keep  the map method and  returned a map rather than  the array from the Promise.all.

Reply:
This works too. So much nicer to access the component info by name. Nice work!